### PR TITLE
Extend the base API to allow other extensions to add their own components.

### PR DESCRIPTION
**Pull request by worksofliam**
### Changes

Adds brand new export to the base API which allows other extensions to add their own installable components.

Components are checked on connection to ensure they are installed on the OS.

### How to test this PR

Examples:

1. Run the test cases
2. Ensure extension starts up with existing components as usual.

### Checklist

* [x] have tested my change
* [ ] have created one or more test cases
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/codefori/vscode-ibmi/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by sebjulliand** [comment]
@worksofliam I tried to make the component's workflow as clear as possible for whoever would like to register a component.
Let me know what you think.

Question: would we want to keep track of which extension registers which component and be able to display a list of all the registered components (in the settings view, for example)?

**Comment by sebjulliand** [comment]
> Question: would we want to keep track of which extension registers which component and be able to display a list of all the registered components (in the settings view, for example)?

Forget it: yes we do.
image

@worksofliam give it a whirl 😊

File: src/components/component.ts [hunk 1]
```diff
@@ -1,61 +1,93 @@
 import IBMi from "../api/IBMi";
-import { CopyToImport } from "./copyToImport";
-import { GetMemberInfo } from "./getMemberInfo";
-import { GetNewLibl } from "./getNewLibl";
 
-export enum ComponentState {
-  NotChecked = `NotChecked`,
-  NotInstalled = `NotInstalled`,
+export const enum ComponentState {
```
Comment (worksofliam) [inline src/components/component.ts:3]: Should we convert this to a `type` so it's easier to export?

**Comment by worksofliam** [comment]
@sebjulliand Okay, I think we need one final test each and then we're good!

**Comment by worksofliam** [comment]
Working good for me - let's merge away!

**Pull request merged**